docs/src/content/docs/ko/guides/site-search.mdx

```diff
@@ -0,0 +1,166 @@
+---
+title: 사이트 검색
+description: Starlight에 내장된 사이트 검색 기능과 이를 사용자 정의하는 방법에 대해 알아보세요.
+sidebar:
+  badge: New
+---
+
+import { Tabs, TabItem } from '@astrojs/starlight/components';
+
+기본적으로 Starlight 사이트에는 정적 사이트를 위한 빠르고 낮은 대역폭을 갖춘 검색 도구인 [Pagefind](https://pagefind.app/)에서 제공하는 전체 텍스트 검색이 포함되어 있습니다.
+
+검색을 활성화하는 데 구성이 필요하지 않습니다. 사이트를 구축하고 배포한 다음, 사이트 헤더의 검색창을 사용하여 콘텐츠를 찾을 수 있습니다.
+
+## 검색결과에서 콘텐츠 숨기기
+
+### 페이지 제외
+
+검색 색인에서 페이지를 제외하려면 페이지 프런트매터에 [`pagefind: false`](/ko/reference/frontmatter/#pagefind)를 추가하세요.
+
+```md title="src/content/docs/not-indexed.md" ins={3}
+---
+title: 검색에서 숨길 콘텐츠
+pagefind: false
+---
+```
+
+### 페이지 일부 제외
+
+Pagefind는 [`data-pagefind-ignore`](https://pagefind.app/docs/indexing/#removing-individual-elements-from-the-index) 속성이 있는 요소 내부의 콘텐츠를 무시합니다.
+
+다음 예시에서 첫 번째 단락은 검색 결과에 표시되지만 `<div>`의 내용은 표시되지 않습니다.
+
+```md title="src/content/docs/partially-indexed.md" ins="data-pagefind-ignore"
+---
+title: 부분적으로 색인이 생성된 페이지
+---
+
+이 텍스트는 검색을 통해 찾을 수 있습니다.
+
+<div data-pagefind-ignore>
+
+이 텍스트는 검색에서 숨겨집니다.
+
+</div>
+```
+
+## 대체 검색 공급자
+
+### Algolia DocSearch
+
+[Algolia의 DocSearch 프로그램](https://docsearch.algolia.com/)에 대한 액세스 권한이 있고, 이를 Pagefind 대신 사용하려는 경우, 공식 Starlight DocSearch 플러그인을 사용할 수 있습니다.
+
+1. `@astrojs/starlight-docsearch`를 설치합니다.
+
+   <Tabs>
+
+   <TabItem label="npm">
+
+   ```sh
+   npm install @astrojs/starlight-docsearch
+   ```
+
+   </TabItem>
+
+   <TabItem label="pnpm">
+
+   ```sh
+   pnpm install @astrojs/starlight-docsearch
+   ```
+
+   </TabItem>
+
+   <TabItem label="Yarn">
+
+   ```sh
+   yarn add @astrojs/starlight-docsearch
+   ```
+
+   </TabItem>
+
+   </Tabs>
+
+2. `astro.config.mjs` 파일의 Starlight [`plugins`](/ko/reference/configuration/#plugins) 구성에 DocSearch를 추가하고 Algolia `appId`, `apiKey`, `indexName`를 전달합니다.
+
+   ```js ins={4,10-16}
+   // astro.config.mjs
+   import { defineConfig } from 'astro/config';
+   import starlight from '@astrojs/starlight';
+   import starlightDocSearch from '@astrojs/starlight-docsearch';
+
+   export default defineConfig({
+   	integrations: [
+   		starlight({
+   			title: 'Site with DocSearch',
```

Comment:
I think it would makes sense to translate the title of the website.

Reply:
I have translated this part.

Thanks for pointing out :)